src/uct/base/uct_mem.c

```diff
@@ -137,11 +141,20 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
 
                 ucs_assert(memh != UCT_MEM_HANDLE_NULL);
                 mem->md       = md;
-                mem->mem_type = md_attr.cap.access_mem_type;
+                mem->mem_type = mem_type;
                 mem->memh     = memh;
                 goto allocated;
 
             }
+
+            if (param->mem_type != UCS_MEMORY_TYPE_HOST) {
+                /* assumes that only MDs are capable of allocating non-host
+                 * memory
+                 */
+                ucs_error("unable to allocated requested memory type");
+                return UCS_ERR_UNSUPPORTED;
+            }
```

Comment:
this should be done in non-MD allocation methods

Reply:
done